pkg/network/utils.go

```diff
@@ -105,39 +106,42 @@ func DiscoverPTPDevices() ([]string, error) {
 	return nics, nil
 }
 
-func GetPhcId(iface string) (phcId string) {
-	var out bytes.Buffer
+func GetPhcId(iface string) string {
+	var err error
+	var id int
+	if id, err = getPTPClockIndex(iface); err != nil {
+		glog.Error(err.Error())
+	} else {
+		return fmt.Sprintf("/dev/ptp%d", id)
+	}
+	return ""
+}
 
+func getPTPClockIndex(iface string) (int, error) {
 	if !ethtoolInstalled() {
-		glog.Errorf("discoverDevices(): ethtool not installed. Cannot grab NIC capabilities")
-		return
+		return 0, fmt.Errorf("discoverDevices(): ethtool not installed. Cannot grab NIC capabilities")
 	}
 
-	ethtoolPath, _ := exec.LookPath("ethtool")
+	// Command to get PTP clock info
+	cmd := exec.Command("ethtool", "-T", iface)
 
-	cmd := exec.Command(ethtoolPath, "-T", iface)
-	cmd.Stdout = &out
-	err := cmd.Run()
+	// Execute the command and capture output
+	out, err := cmd.CombinedOutput()
 	if err != nil {
-		glog.Infof("GetPhcId: could not grab NIC timestamp capability for %s: %v", iface, err)
-		return
+		return -1, fmt.Errorf("failed to run ethtool: %w", err)
 	}
 
-	phcId = parsePhcId(&out)
-	if phcId != "" {
-		return fmt.Sprintf("/dev/%s", phcId)
+	// Regex to extract clock index
+	re := regexp.MustCompile(`PTP Hardware Clock: (\d+)`)
```

Comment:
use _ETHTOOL_PTP_HARDWARE_CLOCK or delete the definition of _ETHTOOL_PTP_HARDWARE_CLOCK if not using it.

Reply:
done